src/routes/hooks/hooks.service.ts

```diff
@@ -0,0 +1,36 @@
+import { IConfigurationService } from '@/config/configuration.service.interface';
+import { IQueuesRepository } from '@/domain/queues/queues-repository.interface';
+import { WebHookSchema } from '@/routes/hooks/entities/schemas/web-hook.schema';
+import { OnModuleInit } from '@nestjs/common';
+import { ConsumeMessage } from 'amqplib';
+import { Event } from '@/routes/hooks/entities/event.entity';
+import { ILoggingService } from '@/logging/logging.interface';
+
+export abstract class HooksService implements OnModuleInit {
```

Comment:
While I'm totally aligned with the purpose of this refactor, I think I don't fully understand this abstract class part: are we planning to extend `HooksService` with another derived class for notifications (e.g.: `HooksNotificationsService`)? 

If that is the case, wouldn't that make `onModuleInit` execute twice, and therefore create two queue subscriptions? Would the `HooksController` dispatch the events to the proper implementation of `HooksService`, or all events would be consumed by all the implementation classes?

Reply:
As per our discussion, I've [opened a different PR](https://github.com/safe-global/safe-client-gateway/pull/1774) that moves the queue subscription into the domain, but accepts multiple listeners for said subscription. This should prevent multiple subscriptions. I will rebase this PR after merging that.